delatore/sources/influx.py

```diff
@@ -577,6 +577,64 @@ async def _get_status(self, metric):
         return disk_statuses
 
 
+
+class InfluxSourceAutoscalling(InfluxSource):
+    """InfluxSourceAutoscallingclient"""
+
+    # pylint: disable=unused-variable
+
+    CONFIG_ID = 'influxdb_autoscalling'
+    _threshold = 30
+    _error_message_template = _err_message_template('error_message_autoscalling.txt')
+
+
+    async def get_update(self) -> dict:
+        main_metric = self._metrics[0]
+        error_status, last_time_ms, status = await self._get_status(main_metric)
+        results = []
+        if status == status.FAIL:
+            results.append(generate_status(main_metric.name, status,
+                                           last_time_ms.strftime(UNIFIED_TIME_PATTERN)))
+        if status == status.NO_DATA:
+            results.append(generate_status(main_metric.name, status, None))
+        else:
+            if error_status == True:
+                auxilary_metric = self._metrics[1]
+                cpu_utilization = await self._get_auxilary_metrics(auxilary_metric)
+                results.append(generate_error(main_metric.name, self._error_message_template.format(threshold=self._threshold, cpu_utilization=cpu_utilization)))
+        return generate_message(self.CONFIG_ID, results)
+
+    async def _get_auxilary_metrics(self, metric):
+        query = metric.query.format(entity=metric.metric_id)
+        result = await self.influx_client.query(query)
+        try:
+            cpu_utilization = result.raw['series'][0]['values'][0][1]
+        except(IndexError, KeyError):
+            cpu_utilization = 'NO DATA'
+        return cpu_utilization
+
+
+    async def _get_status(self, metric):
+        query = metric.query.format(entity=metric.metric_id)
+        last_record = await self.influx_client.query(query)
+        last_time_ms = None
+        error_status = False
+        status = Status.OK
+        try:
+            response_time = last_record.raw['series'][0]['values'][0][1]
```

Comment:
`last_record.raw['series'][0]['values'][0]` is common for two expressions, you can move it into varialbe (and handle exceptions more explicit)